reward/supply_manager.go

```diff
@@ -379,9 +380,35 @@ func (sm *supplyManager) totalSupplyFromState(num uint64) (*big.Int, error) {
 	return totalSupply, nil
 }
 
+func (sm *supplyManager) getAccRewardUncached(num uint64) (*database.AccReward, error) {
+	accReward := sm.db.ReadAccReward(num)
+	if accReward != nil {
+		return accReward, nil
+	}
+
```

Comment:
maybe introduce a fast path by trying `num - num % checkpointInterval`?

Reply:
In theory, checkpointInterval can be different after a node restart because it originates from the node config TrieBlockInterval (`--state.block-interval`). I don't want this to rely on the 128-by-default assumption. Same logic applied in https://github.com/kaiachain/kaia/blob/v1.0.1/node/cn/state_accessor.go#L87.

Reply:
yeah I understand why the logic below must exist, but I think it's very likely to hit the branch if we introduced one before going down all the blocks. It can cause one additional DB read overhead, but a high chance of hitting.

Reply:
Nice idea. Added in c1401375.

Reply:
it's not related to this PR, but ian's comment gives me a thing to consider about block interval. It may occur some problems if node operator changes block interval to unexpected values. For example, it's not safe to use 128-by-default assumption and even TrieBlockInterval since when node startup with updated `TrieBlockInterval`, it might not have states at `num - num%updatedTrieBlockInterval`. Maybe we also need to consider this for our modular architecture.

Reply:
@hyeonLewis Absolutely. We shall not rely on the "fixed block interval" as it is not a universal concept.